src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -370,9 +573,17 @@ class column_store
         return materialize(std::move(it));
     }
 
+    void clear() {
+        // replace *this with an empty instance
+        auto replacement = column_store{};
+        auto current_fields = member_fields();
+        auto new_fields = replacement.member_fields();
+        std::swap(current_fields, new_fields);
+    }
     void prefix_truncate(int64_t bo) {
         auto lb = _base_offset.lower_bound(bo);
         if (lb == _base_offset.end()) {
+            clear();
             return;
         }
```

Comment:
@Lazin clear seems missing here; it was the reason test_segment_meta_cstore_prefix_truncate_full was failing for some inputs. It's probably a branch that would never be taken if _write_buffer wasn't there